webapp/src/components/chat/prompt-dialog/PromptDialog.tsx

```diff
@@ -47,19 +51,32 @@ export const PromptDialog: React.FC<IPromptDialogProps> = ({ message }) => {
     const classes = useClasses();
     const dialogClasses = useDialogClasses();
 
+    const [selectedTab, setSelectedTab] = React.useState<TabValue>('formatted');
+    const onTabSelect: SelectTabEventHandler = (_event, data) => {
+        setSelectedTab(data.value);
+    };
+
     let prompt: string | BotResponsePrompt;
     try {
         prompt = JSON.parse(message.prompt ?? '{}') as BotResponsePrompt;
     } catch (e) {
         prompt = message.prompt ?? '';
     }
+
     let promptDetails;
     if (typeof prompt === 'string') {
-        promptDetails = prompt.split('\n').map((paragraph, idx) => <p key={`prompt-details-${idx}`}>{paragraph}</p>);
+        promptDetails = formatParagraphTextContent(prompt);
     } else {
         promptDetails = Object.entries(prompt).map(([key, value]) => {
             const isStepwiseThoughtProcess = Constants.STEPWISE_RESULT_NOT_FOUND_REGEX.test(value as string);
-            return value ? (
+            if (
+                key === 'chatMemories' &&
+                value &&
+                !(value as string).includes('User has also shared some document snippets:')
```

Comment:
where does this come from? should we rely on string content? wondering if this could change in the future if its coming from the Kernel

Reply:
yeah, this comes from the DocumentMemorySkill in webapi but since there's all the memories are queried together, this is the cleanest way to identify document memories specifically